cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -276,8 +276,14 @@ void generate_depth_remappings(
     }
   }
   auto sync_fn = [](decltype(read_tasks) read_tasks) {
-    for (auto& task : read_tasks) {
-      task.wait();
+    std::vector<std::thread> threads;
+    for (std::size_t task_idx = 0; task_idx < read_tasks.size(); ++task_idx) {
+      threads.emplace_back([](std::future<size_t>& task) { task.wait(); },
```

Comment:
hm, I don't think we want this. We're creating a thread for each read task. Maybe using a single thread to wait on all tasks would be okay, but even that is iffy IMO

Reply:
Thanks! Now that we know the subtasks are submitted eagerly, it is of a trivial issue whether the vector of tasks here are run sequentially or in parallel. So I'll revert the change here.

Reply:
Completed